packages/ui/src/components/Markdown/index.tsx

```diff
@@ -10,19 +10,28 @@ const StyledMarkdown = styled.div`
   a {
     color: ${({theme}) => theme.colors.textSecondary};
   }
-  .mdx a {
+
+  .markdown a {
     padding: 0 ${paddings.small};
   }
 `;
 
-const Markdown = ({children}: {children: React.ReactNode}) => {
-  const mdxComponents = {code: Highlighter};
-
-  return (
-    <MDXProvider components={mdxComponents}>
-      <StyledMarkdown>{children}</StyledMarkdown>
-    </MDXProvider>
+const Markdown = ({source}: {source: string}) => {
+  const elements = ReactHtmlParser(source);
+  const highlighted = elements.map((el: React.ReactElement, idx: number) =>
+    el.type === "pre" && el.props.children.length > 0 ? (
+      <Highlighter
+        className={el.props.children[0].props.className}
+        key={el.key || idx}
```

Comment:
There could be a conflict where an element has a key of `0` here. Might want to prefix it or something to avoid collisions. Something like:

```typescript
        key={el.key ? `key_${el.key}` : idx}
```